### PR TITLE
codecoverage/bot: Move generate_suite_reports from codecov.py to a separate file

### PR DESCRIPTION
**Pull request by rv404674**
Moved the generate_suits_report function from codecov.py to a separate file ( suite_reports)

First part of #1030.

### CODE REVIEW

**Comment by marco-c** [comment]
Hey @rv404674, thanks for working on this! It looks like there are failures on Taskcluster (you can see them by clicking on "Details" and then go to the logs). You can also run the linting and tests locally by running `./please check codecoverage/bot`.

Note that, in order to fully fix #1030, we also need to add some kind of test, but this can be done later.

**Comment by marco-c** [comment]
> Imports are in the right position still i am getting this error.

They aren't in the right position. For example in codecov.py you have:
```
from code_coverage_bot import chunk_mapping
from code_coverage_bot import suite_reports
from code_coverage_bot import grcov
from code_coverage_bot import taskcluster
```

suite_reports should go after grcov and before taskcluster (as they are ordered alphabetically).

**Comment by rv404674** [comment]
Thanks @marco-c , i have added the changes that you suggested. Can you take a look when you get some time, also UT are still remaining.

File: src/codecoverage/bot/code_coverage_bot/codecov.py [hunk 1]
```diff
@@ -166,7 +138,7 @@ def go(self):
         else:
             logger.info('Generating suite reports')
             os.makedirs(self.ccov_reports_dir, exist_ok=True)
-            self.generate_suite_reports()
+            suite_reports.generate_suite_reports(self.suites, self.artifactsHandler, self.ccov_reports_dir, self.repo_dir)
```
Comment (marco-c) [inline src/codecoverage/bot/code_coverage_bot/codecov.py]: The function in suite_reports is not called `generate_suite_reports`, but `generate`.

**Comment by rv404674** [comment]
hey @marco, done. Can you take a final look. UT still remains.

**Comment by marco-c** [comment]
> And can you help me with error

If you rebase (`git pull upstream master --rebase`), the error should go away.

**Comment by rv404674** [comment]
hey @marco-c , it can be merged now. :). Whenever you get some time, do take a look at it.

**Pull request merged**